doc/api/stream.md

```diff
@@ -1859,6 +1859,40 @@ run().catch(console.error);
 after the `callback` has been invoked. In the case of reuse of streams after
 failure, this can cause event listener leaks and swallowed errors.
 
+### `stream.pipelinify(...streams)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `streams` {Stream[]|Iterable[]|AsyncIterable[]|Function[]}
+* Returns: {stream.Duplex}
+
+Combines multiple streams into a `Duplex` stream. This works by
+calling `stream.pipeline` on all the passed streams and then creating
+a `Duplex` stream which writes to the first passed stream and reads from
+the stream returned from `stream.pipeline`.
```

Comment:
I think the underlying _how_ is less important than describing the effective behavior and why `pipelinify` is useful.

```suggestion
Combines two or more streams into a `Duplex` stream that writes to the 
first stream and reads from the last. Each provided stream is piped into
the next, using `stream.pipeline`. If any of the streams error then all
are destroyed, including the outer `Duplex` stream.

Because `stream.pipelinify` returns a new stream that in turn can (and
should) be piped into other streams, it enables composition. In contrast,
when passing streams to `stream.pipeline`, typically the first stream is
a readable stream and the last a writable stream, forming a closed 
circuit.
```